IBM/carbon-addons-iot-react
# chore(deps): update semantic-release@beta for pre-release support

**Summary**

- To support auto deploys of pre-release branches to pre-release channels, ie `carbon-addons-iot-react@beta`, `semantic-release` and associated deps must be upgraded to v16 beta. 

**Change List (commits, features, bugs, etc)**

- update `semantic-release` and associated deps
- update `.travis.yml` deploy step to use beta instead of latest

**Acceptance Test (how to verify the PR)**

It doesn't look like there isn't a way to fully test this end-to-end until we merge this. Same with pre-releases, we'll have to verify it's working when we create the `beta` branch.

You can test a dry run via:
1. Checkout this branch
2. Add the following script to package.json, you'll need a Github token and an NPM token:
```json
"release:dryrun": "NPM_TOKEN={replaceWithYourNpmTokenHere} GH_TOKEN={replaceWithYourGitHubTokenHere} semantic-release --dry-run --no-ci --branches semantic-release-upgrade",
```
3. Run `yarn release:dryrun`

I did this and all was successful. It should correctly determine that there isn't anything "new" to publish and therefore not cut a new version.

```sh
❯ yarn release:dryrun
yarn run v1.12.3
$ NPM_TOKEN=cfd078f9-4ff6-469c-aa3a-2f8c10c1d4ee GH_TOKEN=dc0073041f45c8e3c62e2cbc9f38a6ca90d4642c semantic-release --dry-run --no-ci --branches semantic-release-upgrade
[12:51:55 PM] [semantic-release] › ℹ  Running semantic-release version 16.0.0-beta.19
[12:51:55 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[12:51:55 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[12:51:55 PM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[12:51:55 PM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[12:51:55 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[12:51:55 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[12:51:55 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[12:51:55 PM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/npm"
[12:51:55 PM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/github"
[12:51:55 PM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[12:51:55 PM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[12:52:01 PM] [semantic-release] › ⚠  Run automated release from branch semantic-release-upgrade in dry-run mode
[12:52:05 PM] [semantic-release] › ✔  Allowed to push to the Git repository
[12:52:05 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/npm"
[12:52:05 PM] [semantic-release] [@semantic-release/npm] › ℹ  Verify authentication for registry https://registry.npmjs.org/
[12:52:05 PM] [semantic-release] [@semantic-release/npm] › ℹ  Wrote NPM_TOKEN to /Users/taylorjones/dev/carbon-addons-iot-react/.npmrc
[12:52:06 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/npm"
[12:52:06 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/github"
[12:52:06 PM] [semantic-release] [@semantic-release/github] › ℹ  Verify GitHub authentication
[12:52:07 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/github"
[12:52:07 PM] [semantic-release] › ℹ  Found git tag v1.58.0 associated with version 1.58.0 on branch semantic-release-upgrade
[12:52:07 PM] [semantic-release] › ℹ  Found 1 commits since last release
[12:52:07 PM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[12:52:07 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore(deps): update semantic-release@beta for pre-release support
[12:52:07 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[12:52:07 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analysis of 1 commits complete: no release
[12:52:07 PM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[12:52:07 PM] [semantic-release] › ℹ  There are no relevant changes, so no new version is released.
✨  Done in 12.61s.
```
